source/gfoidl.Stochastics/Statistics/Sample.AutoCorrelation.cs

```diff
@@ -33,21 +53,68 @@ internal IEnumerable<double> AutoCorrelationSimd()
                 yield return r_xx / (arr.Length - m);
             }
         }
+        #endregion
         //---------------------------------------------------------------------
-        internal IEnumerable<double> AutoCorrelationSequential()
+        #region ToArray
+        internal double[] AutoCorrelationToArraySimd()
         {
-            double[] arr = _values;
-            int n2       = arr.Length >> 1;
+            int n    = _values.Length;
+            int n2   = n >> 1;
+            var corr = new double[n2];
 
-            for (int m = 0; m < n2; ++m)
+            this.AutoCorrelationToArrayImpl(corr, (0, n2));
+
+            return corr;
+        }
+        //---------------------------------------------------------------------
+        internal double[] AutoCorrelationToArrayParallelSimd()
+        {
+            var corr = new double[_values.Length / 2];
+
+            Parallel.ForEach(
+                Partitioner.Create(0, _values.Length / 2),
```

Comment:
`Partitioner.Create` produces equal sized ranges, but the workload is decreasing as `m` increments in the outer loop, so that the inner loop has less work to do `for (int k = m; k < arr.Length; ++k)`. Thus the workload is not even distributed among the cores.

There should be a better partitioning, so the perf-win will be greater. The values for the threshold should be remeasured after that.

Reply:
Is addressed in https://github.com/gfoidl/Stochastics/issues/9